source-countercyclical contribution from williamleiby

williamleiby opened the pull request:
## What

This PR adds a new connector for source Countercyclical (source-countercyclical).

Contributor-provided connector description:

Countercyclical is the fully end-to-end financial intelligence platform designed for modern investment teams.
## Details

<details><summary>Expand for list of spec properties and streams</summary>

### Configuration

| Input | Type | Description | Default Value |
|-------|------|-------------|---------------|
| `api_key` | `string` | API Key.  |  |

### Streams
| Stream Name | Primary Key | Pagination | Supports Full Sync | Supports Incremental |
|-------------|-------------|------------|---------------------|----------------------|
| Investments | id | No pagination | ✅ |  ❌  |
| Valuations | id | No pagination | ✅ |  ❌  |
| Memos | id | No pagination | ✅ |  ❌  |

</details>

## Reviewer checklist
- [ ] Run `/format-fix` to resolve any formatting errors
- [ ] Help the contributor upload an icon that follows our icon design guidelines
- [ ] Click into the CI workflows that wait for a maintainer to run them, which should trigger CI runs

<!-- DO NOT REMOVE: connector-builder-new-connector-contribution -->

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/airbytehq/airbyte?pullRequest=46530) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/airbytehq/airbyte?pullRequest=46530) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/airbytehq/airbyte?pullRequest=46530) it.</sub>

natikgadzhi reviewed (commented):
Thank you for this! 

@williamleiby, Countercyclical [supports limit/offset pagination](https://docs.countercyclical.io/developers/pagination). Can you add that and re-create the PR? 

Alternatively, I can merge this (once you add the icon), and you can do a follow-up PR with the pagination fixes. 

Let me know what works better!

williamleiby commented:
> Thank you for this!
> 
> @williamleiby, Countercyclical [supports limit/offset pagination](https://docs.countercyclical.io/developers/pagination). Can you add that and re-create the PR?
> 
> Alternatively, I can merge this (once you add the icon), and you can do a follow-up PR with the pagination fixes.
> 
> Let me know what works better!

Hello @natikgadzhi , thank you for the quick response!

We can go ahead and merge this (once I update the icon) as our docs didn't appear to be fully reflective of the `v1` endpoints/streams we added in here (e.g. `/investments`, etc.), my apologies about that! 

Updated those docs just now (see here for [investments](https://docs.countercyclical.io/developers/endpoints/investments), [valuations](https://docs.countercyclical.io/developers/endpoints/valuations), and [memos](https://docs.countercyclical.io/developers/endpoints/memos)) & will follow up with any necessary PRs to make sure there aren't any changes that would break this connector. Thanks again! Let me know if there's anything else, and I appreciate your help!

natikgadzhi commented:
Alright! This will likely automerge tonight. Thank you for a new connector! 👏🏼 

Once it shows up in the Cloud registry, you can open it in Builder and edit it, add pagination to your streams.
